model/src/main/scala/aqua/model/func/FuncCallable.scala

```diff
@@ -129,9 +146,24 @@ case class FuncCallable(
             Eval.now(Chain.empty)
           )
       }
-      .map { case ((_, resolvedExports), b) =>
+      .map { case ((_, resolvedExports), callableFuncBody) =>
         // If return value is affected by any of internal functions, resolve it
-        FuncOp(b) -> result.map(_.resolveWith(resolvedExports))
+        (for {
+          exp <- call.exportTo
+          res <- result
+          pair <- exp match {
+            // TODO check variable, not string
+            case Call.Export(name, StreamType(_)) if name == "-return-" =>
```

Comment:
Where does this case come from?

Reply:
all will be wrapped with `identity result -return-`, if result is a stream